src/vanilla/Model/CompositeTypeCs.cs

```diff
@@ -159,13 +159,29 @@ public string Signature
                     foreach (PropertyCs property in Parameters.Where(p => !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty))
                     {
                         string format = (property.IsRequired ? "{0} {1}" : "{0} {1} = default({0})");

```

Comment:
why does this change so much? I'd have expected signatures to look identical to what we did for `modelAsString: false` before

Reply:
For extensible enums we are generating the parameter type based on the information in `EnumType`, `property.ModelTypeName` does not match up with the name of the extensilbe enum class generated hence the change.
Also, we are using extensible enums only if modelAsString is also set to true